### PR TITLE
feat(services): extract dockerfile settings for lifecycle job

### PR DESCRIPTION
**Pull request by ctjhoa**
# What does this PR do?

https://qovery.atlassian.net/browse/FRT-1153

### CODE REVIEW

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview cc1de7de-94e5-40f8-8e45-bc8986d9dfec | [storybook](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/cc1de7de-94e5-40f8-8e45-bc8986d9dfec/services/general) |
 /qovery []() preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> Truth can only be found in one place: the code

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

A preview environment was automatically created via Qovery.
Click on the link below to follow its deployment and use it.
:point_right: [\[PR\] test-remi2 - Frt 1153 pimp lifecycle job add raw docker - 2024-06-18T12:26:50Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/e773b890-e6b2-4406-93a9-71f9a7ab6d73/environment/c3163726-8b98-4536-80d1-3ad1f360de9b/services/general)

 Another comment will be posted when deployments are finished

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Your preview environment failed to be deployed !
Click on the link below to check the deployment logs.
:point_right: [\[PR\] test-remi2 - Frt 1153 pimp lifecycle job add raw docker - 2024-06-18T12:26:50Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/e773b890-e6b2-4406-93a9-71f9a7ab6d73/environment/c3163726-8b98-4536-80d1-3ad1f360de9b/logs)

File: libs/domains/services/feature/src/lib/dockerfile-raw-modal/dockerfile-raw-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+          </BlockContent>
+          <BlockContent
+            title="Default Dockerfile"
+            className="mb-0 rounded-l-none"
+            classNameContent="p-0"
+            headRight={
+              <CopyToClipboardButtonIcon className="text-xs hover:text-neutral-400" content={defaultContent} />
+            }
+          >
+            <CodeEditor width="100%" height="calc(100vh  - 254px)" defaultValue={defaultContent} readOnly />
```
Comment (RemiBonnet) [inline libs/domains/services/feature/src/lib/dockerfile-raw-modal/dockerfile-raw-modal.tsx:52]:
```suggestion
            <CodeEditor width="100%" height="calc(100vh  - 254px)" defaultValue={defaultContent} language="dockerfile" readOnly />
```

Allow colorization for Dockerfile

File: libs/pages/application/src/lib/feature/page-settings-dockerfile-feature/__snapshots__/page-settings-dockerfile-feature.spec.tsx.snap [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`PageSettingsDockerfileFeature should match snapshot 1`] = `
+<body>
+  <div />
+</body>
+`;
```
Comment (RemiBonnet) [inline libs/pages/application/src/lib/feature/page-settings-dockerfile-feature/__snapshots__/page-settings-dockerfile-feature.spec.tsx.snap:7]: Missing something to generate snapshot ?

File: libs/domains/services/feature/src/lib/dockerfile-settings/dockerfile-settings.tsx [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+                  name={field.name}
+                  onChange={field.onChange}
+                  value={field.value ?? ''}
+                  label="Dockerfile path"
+                  error={error?.message}
+                  hint={
+                    <>
+                      <span>Specify the location of your dockerfile. Expected format: myapp/Dockerfile</span>
+                      <br />
+                      <ExternalLink>Create one with Docker init</ExternalLink>
```
Comment (RemiBonnet) [inline libs/domains/services/feature/src/lib/dockerfile-settings/dockerfile-settings.tsx:153]:
```suggestion
                      <ExternalLink size="xs">Create one with Docker init</ExternalLink>
```

File: libs/pages/services/src/lib/feature/page-job-create-feature/step-dockerfile-feature/step-dockerfile-feature.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  SERVICES_JOB_CREATION_GENERAL_URL,
+  SERVICES_URL,
+} from '@qovery/shared/routes'
+import { Button, FunnelFlowBody, Heading, Section } from '@qovery/shared/ui'
+import { useDocumentTitle } from '@qovery/shared/util-hooks'
+import { buildGitRepoUrl } from '@qovery/shared/util-js'
+import { useJobContainerCreateContext } from '../page-job-create-feature'
+
+export function StepDockerfileFeature() {
+  useDocumentTitle('Configure - Create Job')
```
Comment (RemiBonnet) [inline libs/pages/services/src/lib/feature/page-job-create-feature/step-dockerfile-feature/step-dockerfile-feature.tsx:17]:
```suggestion
  useDocumentTitle('Dockerfile - Create Job')
```

File: libs/pages/services/src/lib/ui/page-job-create/step-summary/step-summary.tsx [hunk 5]
```diff
@@ -140,6 +143,30 @@ export function StepSummary(props: StepSummaryProps) {
...
+              <ul className="flex list-none flex-col gap-2 text-sm text-neutral-400">
+                {props.dockerfileData.dockerfile_path && (
+                  <li>
+                    <strong className="font-medium">Dockerfile path:</strong> {props.dockerfileData.dockerfile_path}
+                  </li>
+                )}
+                {props.dockerfileData.dockerfile_raw && (
+                  <li>
+                    <strong className="font-medium">From raw Dockerfile:</strong> {}
+                    {truncateText(props.dockerfileData.dockerfile_raw, 50)}
```
Comment (RemiBonnet) [inline libs/pages/services/src/lib/ui/page-job-create/step-summary/step-summary.tsx:163]: Missing "..." at the end of the truncate maybe
```suggestion
                    {truncateText(props.dockerfileData.dockerfile_raw, 50)}...
```

File: package.json [hunk 6]
```diff
@@ -192,5 +192,6 @@
         "@semantic-release/github"
       ]
     ]
-  }
+  },
+  "packageManager": "yarn@3.6.1+sha256.811210abb5fb5751da12ead8a9cbc0c150b07e43ac9cbedec6752d22abfd2bd6"
```
Comment (RemiBonnet) [inline package.json:196]: necessary?

**Review by RemiBonnet — CHANGES_REQUESTED** [review]
Thanks for this PR !

The behavior to save an change on the file is different with `Raw YAML`, could you harmonize it ? 

**Raw YAML:**
Screenshot 2024-06-19 at 10 40 06

**Raw Dockerfile:** with save behavior not directly on the modal
Screenshot 2024-06-19 at 10 40 51

File: package.json (hunk 6 above)
Comment (ctjhoa) [reply package.json]: My bad, newer yarn cli tends to add this automatically. Removed
https://github.com/Qovery/console/pull/1433/commits/ac99ffdd49c46ffaa5d58b45f33315f67a19329d

File: libs/pages/services/src/lib/ui/page-job-create/step-summary/step-summary.tsx (hunk 5 above)
Comment (ctjhoa) [reply libs/pages/services/src/lib/ui/page-job-create/step-summary/step-summary.tsx]: Fixed
https://github.com/Qovery/console/pull/1433/commits/ac99ffdd49c46ffaa5d58b45f33315f67a19329d

File: libs/pages/services/src/lib/feature/page-job-create-feature/step-dockerfile-feature/step-dockerfile-feature.tsx (hunk 4 above)
Comment (ctjhoa) [reply libs/pages/services/src/lib/feature/page-job-create-feature/step-dockerfile-feature/step-dockerfile-feature.tsx]: Fixed
https://github.com/Qovery/console/pull/1433/commits/ac99ffdd49c46ffaa5d58b45f33315f67a19329d

File: libs/domains/services/feature/src/lib/dockerfile-settings/dockerfile-settings.tsx (hunk 3 above)
Comment (ctjhoa) [reply libs/domains/services/feature/src/lib/dockerfile-settings/dockerfile-settings.tsx]: Fixed
https://github.com/Qovery/console/pull/1433/commits/ac99ffdd49c46ffaa5d58b45f33315f67a19329d

File: libs/domains/services/feature/src/lib/dockerfile-raw-modal/dockerfile-raw-modal.tsx (hunk 1 above)
Comment (ctjhoa) [reply libs/domains/services/feature/src/lib/dockerfile-raw-modal/dockerfile-raw-modal.tsx]: Ok I thought dockerfile wasn't supported but it is ! Thanks for pointing it out, fixed
https://github.com/Qovery/console/pull/1433/commits/ac99ffdd49c46ffaa5d58b45f33315f67a19329d

File: libs/pages/application/src/lib/feature/page-settings-dockerfile-feature/__snapshots__/page-settings-dockerfile-feature.spec.tsx.snap (hunk 2 above)
Comment (ctjhoa) [reply libs/pages/application/src/lib/feature/page-settings-dockerfile-feature/__snapshots__/page-settings-dockerfile-feature.spec.tsx.snap:224]: https://github.com/Qovery/console/pull/1433/commits/b6deb42b3b9817f053b94d9878abb355eb91b189

**Review by RemiBonnet — APPROVED** [review]

**Pull request merged**